pkg/errors/errors_test.go

```diff
@@ -23,3 +38,2006 @@ func TestErrorFind(t *testing.T) {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(tt.fields.err.IsMaxLimitForServiceAccountReached()).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_IsBadRequest(t *testing.T) {
+	type fields struct {
+		err *ServiceError
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "should return a boolean value true if the error IsBadRequest()",
+			fields: fields{
+				err: &ServiceError{},
+			},
+			want: e.Code == BadRequest(genericErrorMessage).Code,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(tt.fields.err.IsBadRequest()).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_InSufficientQuota(t *testing.T) {
+	type fields struct {
+		err *ServiceError
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "should return a boolean value true if the error is InSufficientQuota()",
+			fields: fields{
+				err: &ServiceError{},
+			},
+			want: e.Code == InsufficientQuotaError(genericErrorMessage).Code,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(tt.fields.err.InSufficientQuota()).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_IsFailedToCheckQuota(t *testing.T) {
+	type fields struct {
+		err *ServiceError
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "should return a boolean value true if the error IsFailedToCheckQuota()",
+			fields: fields{
+				err: &ServiceError{},
+			},
+			want: e.Code == FailedToCheckQuota(genericErrorMessage).Code,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(tt.fields.err.IsFailedToCheckQuota()).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_IsInstanceTypeNotSupported(t *testing.T) {
+	type fields struct {
+		err *ServiceError
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "should return a boolean value true if the error IsInstanceTypeNotSupported()",
+			fields: fields{
+				err: &ServiceError{},
+			},
+			want: e.Code == InstanceTypeNotSupported(genericErrorMessage).Code,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(tt.fields.err.IsInstanceTypeNotSupported()).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_IsClientErrorClass(t *testing.T) {
+	type fields struct {
+		err *ServiceError
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   bool
+	}{
+		{
+			name: "should return a boolean value true if the error IsClientErrorClass()",
+			fields: fields{
+				err: &ServiceError{},
+			},
+			want: e.HttpCode >= http.StatusBadRequest && e.HttpCode < http.StatusInternalServerError,
```

Comment:
instead of copying logic from the tested function, could you instead add expected boolean value? Could you also add test cases for various scenarios possible based on the boolean check?